### PR TITLE
Add Travis CI, refactor gulp tasks

### PR DESCRIPTION
**Pull request by XhmikosR**
@feimosi: do not merge this yet, we need to decide which node.js version to run tests on.

In order to see this in action please add Travis CI to this repository.

### CODE REVIEW

**Comment by XhmikosR** [comment]
So which Node.js version you want to use @feimosi?

**Comment by XhmikosR** [comment]
OK, I think now it's ready.

**Comment by XhmikosR** [comment]
@feimosi: *bump*

I also added a fix for autoprefixer; previously you only specified FF 3.6 so there were missing prefixes for FF.

**Comment by feimosi** [comment]
I'll review and merge this, as well as #93, during the coming weekend.

**Comment by XhmikosR** [comment]
Cool, I wouldn't merge 93 personally, not close to a stable at least.

File: gulpfile.js [hunk 1]
```diff
@@ -97,12 +97,14 @@ gulp.task('update-version', function () {
             dist.js + '*.js'
             ], {
```
Comment (feimosi) [inline gulpfile.js]: Could you export this array of globs to a variable above the return? It'll be more readable.

File: gulpfile.js [hunk 2]
```diff
@@ -137,4 +139,8 @@ gulp.task('patch', function() {
     runSequence('bump-patch', 'build', 'update-version');
 });
 
-gulp.task('build', ['css', 'js', 'css-min', 'js-min']);
+gulp.task('build', function() {
+    runSequence('lint', 'css', 'js', 'css-min', 'js-min', 'update-version')
```
Comment (feimosi) [inline gulpfile.js]: We've talked about that before: why do you use _runSequence_ like this? `css` and `css-min` as well as `js` and `js-min` are independent and can freely run in parallel.

Comment (XhmikosR) [inline gulpfile.js]: Well, like I have said, I'm not familiar with Gulp. Adn css-min depends on css the same as js-min depends on js.

File: gulpfile.js (hunk 1 above)
Comment (XhmikosR) [inline gulpfile.js]: Feel free to do it later after we are done merging the PR.

File: gulpfile.js (hunk 2 above)
Comment (feimosi) [inline gulpfile.js]: How can you say they depend on each other when I see they don't? I've worked with Gulp long enough. Show me the exact lines.

Comment (XhmikosR) [inline gulpfile.js]: So, you want to minify something before that task has finished?

Comment (feimosi) [inline gulpfile.js]: They both do the same thing. I know it's suboptimal and could be rewritten, but for now, they are independent - one processes files and puts them inside _build/_ whilst the second one processes files, minifies them and puts in _dist/_

Comment (XhmikosR) [inline gulpfile.js]: Just paste what you suggest then.

Comment (feimosi) [inline gulpfile.js]:
```js
runSequence('lint', ['css', 'js', 'css-min', 'js-min'], 'update-version')
```
this should run the array in parallel

**Comment by XhmikosR** [comment]
Done.

**Comment by XhmikosR** [comment]
BTW, it seems the accessibility fixes made HTML_Codesniffer [pass](https://travis-ci.org/mpc-hc/mpc-hc.org/builds/121795961) too :D

File: gulpfile.js [hunk 3]
```diff
@@ -137,4 +151,8 @@ gulp.task('patch', function() {
     runSequence('bump-patch', 'build', 'update-version');
 });
 
-gulp.task('build', ['css', 'js', 'css-min', 'js-min']);
+gulp.task('build', function() {
+    runSequence('lint', ['css', 'js', 'css-min', 'js-min'], 'update-version');
+});
+
+gulp.task('test', ['build', 'lint']);
```
Comment (feimosi) [inline gulpfile.js]: `build` task already invokes `lint`. It should stay only with `test`.

Comment (XhmikosR) [inline gulpfile.js]: How about we remove lint from build since it doesn't really belong there?

Comment (XhmikosR) [inline gulpfile.js]: Nevermind, I think we say the same thing :P

**Comment by XhmikosR** [comment]
Done.

**Comment by feimosi** [comment]
LGTM :+1:

**Comment by XhmikosR** [comment]
Please do not squash these when merging. They are separate patches and should be like that when merged.

**Comment by feimosi** [comment]
Ok, just please, make sure to rebase the whole branch on top of master and I'll merge it.

**Comment by XhmikosR** [comment]
It is rebased already.

**Comment by feimosi** [comment]
Ok, everything's fine :+1:

**Comment by XhmikosR** [comment]
See that is what I was saying all this time...

**Comment by feimosi** [comment]
Look at my other comment:
https://github.com/feimosi/baguetteBox.js/commit/2d5068461f0d21f091a31cb865289035a9bd784f

**Comment by XhmikosR** [comment]
BTW I think we need margin for figure after all. I'll test it tomorrow and
make a pr until I manage to properly fix the styles.

The regression is on mobile.
On Apr 10, 2016 00:46, "Marek Grzybek" <notifications@github.com> wrote:

Look at my other comment:
2d50684
<https://github.com/feimosi/baguetteBox.js/commit/2d5068461f0d21f091a31cb865289035a9bd784f>

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/feimosi/baguetteBox.js/pull/91#issuecomment-207858647>

**Comment by XhmikosR** [comment]
Also can you make the gh-pages branch and maybe create a dev one? I want to
start working on 2.x.

**Comment by feimosi** [comment]
`gh-pages` is done, as well as `dev` branch. Could you test the current code base yourself, just to make sure and I'm gonna finally make the 1.6.0 release.
Unfortunately, my availability will be limited now.

**Comment by XhmikosR** [comment]
I think everything is fine now.

**Pull request merged**